File: gui/src/quanto/core/CoreException.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package quanto.core;

/**
 *
 * @author alex
 */
public class CoreException extends Exception {
	private static final long serialVersionUID = 1053659906558198953L;
	public CoreException() {
		super();
	}
	public CoreException(String msg) {
		super(msg);
	}
	public CoreException(Throwable cause) {
		super(cause);
	}
	public CoreException(String msg, Throwable cause) {
		super(msg, cause);
	}
}

File: scala/src/main/scala/quanto/core/CoreException.scala
package quanto.core

class CoreException(val message : String = null, val cause: Throwable = null)
extends Exception(message, cause)

class CoreProtocolException(override val message : String = null, override val cause: Throwable = null)
extends CoreException(message, cause)

class CoreUserException(override val message : String, val code: Int)
extends CoreException(message, null)

